packages/purpose-process/test/testGetRiskAnalysisConfigurationByVersion.ts

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import {
+  getMockEService,
+  getMockTenant,
+  writeInReadmodel,
+} from "pagopa-interop-commons-test/index.js";
+import {
+  EServiceId,
+  TenantId,
+  eserviceMode,
+  generateId,
+  tenantKind,
+  toReadModelEService,
+} from "pagopa-interop-models";
+import { describe, expect, it } from "vitest";
+import { genericLogger, getFormRulesByVersion } from "pagopa-interop-commons";
+import {
+  riskAnalysisConfigVersionNotFound,
+  eserviceNotFound,
+  tenantKindNotFound,
+  tenantNotFound,
+} from "../src/model/domain/errors.js";
+import {
+  eservices,
+  purposeService,
+  tenants,
+} from "./purposeService.integration.test.js";
+export const testGetRiskAnalysisConfigurationByVersion = (): ReturnType<
+  typeof describe
+> =>
+  describe("retrieveRiskAnalysisConfigurationByVersion", async () => {
+    it.each(Object.values(tenantKind))(
+      "should retrieve risk analysis configuration by version from the consumer (Eservice mode: deliver) with kind %s",
+      async (kind) => {
+        const mockEservice = {
+          ...getMockEService(),
+          mode: eserviceMode.deliver,
+        };
+        const mockTenant = {
+          ...getMockTenant(),
+          kind,
+        };
+        await writeInReadmodel(toReadModelEService(mockEservice), eservices);
+        await writeInReadmodel(mockTenant, tenants);
+
+        const riskAnalysisVersion = "1.0";
+
+        const result =
+          await purposeService.retrieveRiskAnalysisConfigurationByVersion({
+            eserviceId: mockEservice.id,
+            riskAnalysisVersion,
+            organizationId: mockTenant.id,
+            logger: genericLogger,
+          });
+
+        expect(result).toEqual(
+          getFormRulesByVersion(kind, riskAnalysisVersion)
+        );
+      }
+    );
+    it.each(Object.values(tenantKind))(
+      "should retrieve risk analysis configuration by version from the producer (Eservice mode: receive) with kind %s",
+      async (kind) => {
+        const mockEservice = {
+          ...getMockEService(),
+          mode: eserviceMode.receive,
+        };
+        const mockTenant = {
+          ...getMockTenant(mockEservice.producerId),
+          kind,
+        };
```

Comment:
It my opinion it would be clearer in this way:
```
const mockTenant = {
    ...getMockTenant(),
    kind,
};

const mockEservice = {
    ...getMockEService(),
    mode: eserviceMode.receive,
    producerId: mockTenant.id,
}
```
Anyways, it's the same, as you prefer.

Reply:
I prefer to leave it like this and use the `getMock` in order to set the `producerId`!